applications/StructuralMechanicsApplication/custom_constitutive/elastic_isotropic_3d.cpp

```diff
@@ -97,14 +97,25 @@ void ElasticIsotropic3D::CalculateMaterialResponsePK1 (ConstitutiveLaw::Paramete
 void ElasticIsotropic3D::CalculateMaterialResponseKirchhoff (ConstitutiveLaw::Parameters& rValues)
 {
     CalculateMaterialResponsePK2(rValues);
+
+    //TODO: needs to do the transoformation here!!
 }
 
 /***********************************************************************************/
 /***********************************************************************************/
 
 void ElasticIsotropic3D::CalculateMaterialResponseCauchy (ConstitutiveLaw::Parameters& rValues)
 {
+    //first compute the PK2
```

Comment:
@RiccardoRossi this is small strain. The hypothesis says all the stresses are equal (I remember you the original implementation is yours. If you want to extend it for finite strain, this is called hypoelasticity, the law is correct withy the small strain hypothesis)

Reply:
BTW, the transformation are implemented in the base CL, it is no supposed to be implemented manually each time

Reply:
well, i disagree (partially)

in the hypothesis of small deformations but large rotations you can assume that

~~~
Spk2 = C * Egreen_lagrange 
~~~

and that

~~~
1/detF * F S F^t = sigma_cauchy = C * eps_almansi 
~~~

(with the same C in the two cases)

this is so since

Reply:
Well, in that case we need to redefine many things, as we have been doing everything assuming that we work in small deformations (and also until now small rotations), meaning that we need to redefine all current implementations. I propose to add a flag for large rotations (or assume this is the case only when F is provided, where we can assume that we intend to work with TL or UL) and add the proper calculation only in that case. This way current tests will work, and you will be able to proceed with your calculations. The second option (F provided only) only will enforce us to update a few tests (or even none, the failing ones are for SD element)

Reply:
in any case we could do the push_forward in the  base_solid_element CalculateOnIntegration points function ... but i still think that if you tell to the CL to give a cauchy stress and to compute the strain, it should do it correctly.

Reply:
btw, as of right now, what CL would u use for large rotations?

Reply:
I think my second proposal is better (compute only for F provided), and closer to what you expect

Reply:
i think there is essentially no change, in the sense that right now if  

~~~
USE_ELEMENT_PROVIDED_STRAIN==false 
~~~

than the GreenLagrange strain is computed (btw, note that the name of the strain is green-lagrange NOT cauchy-green, so the name of the function doing it is misleading

one possible solution would be that if you ask for Cauchy stresses than the almansi strain is computed instead of the 
cauchy-green one. the two strains would coincide in the small strain case.